### PR TITLE
Adds cross-env and fixes an issue with getSelf that could occur directly after logging in

### PR DESCRIPTION
**Pull request by poacher2k**
Ref #5

### CODE REVIEW

**Comment by poacher2k** [comment]
Definitively!

This will make it so that the extension only works for logged in users, which might be okay ?

**Comment by timche** [comment]
Yes.

Currently also looking deeper into the #5. Does this actually solve the issue? The feature using `getSelf` still executes [`setFeatureAttribute`](https://github.com/faceit-enhancer/faceit-enhancer/blob/master/src/content/features/add-header-level-progress.js#L19-L22) because `.main-header__right` is also being used on the not logged in site. Once the attribute is set, the feature will not run again (I highly assume). Shouldn't we bail out instead when the user is not logged in? There is a class called `.main-header__right__logged-out` which we can look up first in this feature.

screen shot 2018-05-15 at 11 46 26

**Comment by poacher2k** [comment]
I wasn't able to replicate it after returning null early in `getSelf`, but I agree that it makes more sense to check for `.main-header__right__logged-out` before even calling it instead. As far as I could see, all features using `getSelf` were affected by this, but adding the check to those components is no big task. I'll look into this after work, thanks for the feedback!

**Comment by timche** [comment]
Maybe we can just create a helper function called `isLoggedIn()` and use it where needed. Otherwise the other features using `getSelf` are not called when the required DOM element is not there which I think most of them only occur when the user is logged in. `.main-header__right` might be a rare exception.

**Comment by poacher2k** [comment]
Seems to be working well now!

File: src/content/libs/utils.js [hunk 1]
```diff
@@ -22,3 +23,13 @@ export const notifyIf = async (option, message) => {
     })
   }
 }
+
+export const isLoggedIn = () => {
+  const mainHeaderRightLoggedOut = select('.main-header__right__logged-out')
```
Comment (timche) [inline src/content/libs/utils.js]: `select-dom` has a function called `select.exist` which we could use instead: https://www.npmjs.com/package/select-dom#selectexistsselector-parent--document

**Comment by poacher2k** [comment]
> What exactly do u mean?

`getSelf` would fail because of the missing token, and also be memoized since we return `null`. if you then tried to check the sidebar match elo, that wouldn't work because the failed memoized call wouldn't have any match history, resulting in something like this:

image

after refresh:

image


Adding the `isLoggedIn` check makes it so this won't occur anymore.

**Comment by timche** [comment]
Ah, I get it now. But there are no changes made in `add-sidebar-matches-elo-points` to use `isLoggedIn`(?)

**Comment by poacher2k** [comment]
> But there are no changes made in add-sidebar-matches-elo-points to use isLoggedIn (?)
As far as I could see, none of the other features that use `getSelf`, including add-sidebar-matches-elo-points, would ever get the chance to make the call to `getSelf` before the token was in place, so I don't think it's necessary. Can of course add it for good measure though ?

**Comment by timche** [comment]
No, I'm just a little confused 😄 

> The reason `add-sidebar-matches-elo-points` would stop working, was a result of the failed `getSelf` being memoized. **That is now gone as well :)**

I just don't understand how that is now gone when are no related changes 😄

**Comment by poacher2k** [comment]
> I just don't understand how that is now gone when are no related changes 

I'm just wording this badly ! Failed `getSelf`-calls will still be memoized - it just won't happen anymore, hopefully :stuck_out_tongue:

**Comment by timche** [comment]
Ah, I got it now :+1: Feel free to merge :)

**Comment by poacher2k** [comment]
Thanks for the tip!

**Pull request merged**